{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "ccf18d49-53be-4944-aad7-6a57f67e75dc",
   "metadata": {
    "editable": true,
    "slideshow": {
     "slide_type": ""
    },
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "  0%|                                                                        | 0/20 [00:00<?, ?it/s]\n",
      "  0%|                                                                        | 0/20 [00:00<?, ?it/s]\n",
      "  0%|                                                                        | 0/20 [00:00<?, ?it/s]\n",
      "Traceback (most recent call last):\n",
      "  File \"<string>\", line 1, in <module>\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\spawn.py\", line 120, in spawn_main\n",
      "    exitcode = _main(fd, parent_sentinel)\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\spawn.py\", line 129, in _main\n",
      "    prepare(preparation_data)\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\spawn.py\", line 240, in prepare\n",
      "    _fixup_main_from_path(data['init_main_from_path'])\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\spawn.py\", line 291, in _fixup_main_from_path\n",
      "    main_content = runpy.run_path(main_path,\n",
      "                   ^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"<frozen runpy>\", line 291, in run_path\n",
      "  File \"<frozen runpy>\", line 98, in _run_module_code\n",
      "  File \"<frozen runpy>\", line 88, in _run_code\n",
      "  File \"C:\\Users\\antig\\Documents\\GitHub\\deepALplus\\demo.py\", line 96, in <module>\n",
      "    strategy.train(model_name = args_input.ALstrategy)\n",
      "  File \"C:\\Users\\antig\\Documents\\GitHub\\deepALplus\\query_strategies\\strategy.py\", line 41, in train\n",
      "    self.net.train(labeled_data, X_labeled, Y_labeled,X_unlabeled, Y_unlabeled)\n",
      "  File \"C:\\Users\\antig\\Documents\\GitHub\\deepALplus\\nets_waal.py\", line 56, in train\n",
      "    for index, label_x, label_y, unlabel_x, _ in loader_tr:\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\torch\\utils\\data\\dataloader.py\", line 439, in __iter__\n",
      "    return self._get_iterator()\n",
      "           ^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\torch\\utils\\data\\dataloader.py\", line 387, in _get_iterator\n",
      "    return _MultiProcessingDataLoaderIter(self)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\torch\\utils\\data\\dataloader.py\", line 1040, in __init__\n",
      "    w.start()\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\process.py\", line 121, in start\n",
      "    self._popen = self._Popen(self)\n",
      "                  ^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\context.py\", line 224, in _Popen\n",
      "    return _default_context.get_context().Process._Popen(process_obj)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\context.py\", line 336, in _Popen\n",
      "    return Popen(process_obj)\n",
      "           ^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\popen_spawn_win32.py\", line 45, in __init__\n",
      "    prep_data = spawn.get_preparation_data(process_obj._name)\n",
      "                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\spawn.py\", line 158, in get_preparation_data\n",
      "    _check_not_importing_main()\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\spawn.py\", line 138, in _check_not_importing_main\n",
      "    raise RuntimeError('''\n",
      "RuntimeError: \n",
      "        An attempt has been made to start a new process before the\n",
      "        current process has finished its bootstrapping phase.\n",
      "\n",
      "        This probably means that you are not using fork to start your\n",
      "        child processes and you have forgotten to use the proper idiom\n",
      "        in the main module:\n",
      "\n",
      "            if __name__ == '__main__':\n",
      "                freeze_support()\n",
      "                ...\n",
      "\n",
      "        The \"freeze_support()\" line can be omitted if the program\n",
      "        is not going to be frozen to produce an executable.\n",
      "\n",
      "  0%|                                                                        | 0/20 [00:00<?, ?it/s]\n",
      "  0%|                                                                        | 0/20 [00:00<?, ?it/s]\n",
      "Traceback (most recent call last):\n",
      "  File \"<string>\", line 1, in <module>\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\spawn.py\", line 120, in spawn_main\n",
      "    exitcode = _main(fd, parent_sentinel)\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\spawn.py\", line 129, in _main\n",
      "    prepare(preparation_data)\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\spawn.py\", line 240, in prepare\n",
      "    _fixup_main_from_path(data['init_main_from_path'])\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\spawn.py\", line 291, in _fixup_main_from_path\n",
      "    main_content = runpy.run_path(main_path,\n",
      "                   ^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"<frozen runpy>\", line 291, in run_path\n",
      "  File \"<frozen runpy>\", line 98, in _run_module_code\n",
      "  File \"<frozen runpy>\", line 88, in _run_code\n",
      "  File \"C:\\Users\\antig\\Documents\\GitHub\\deepALplus\\demo.py\", line 96, in <module>\n",
      "    strategy.train(model_name = args_input.ALstrategy)\n",
      "  File \"C:\\Users\\antig\\Documents\\GitHub\\deepALplus\\query_strategies\\strategy.py\", line 41, in train\n",
      "    self.net.train(labeled_data, X_labeled, Y_labeled,X_unlabeled, Y_unlabeled)\n",
      "  File \"C:\\Users\\antig\\Documents\\GitHub\\deepALplus\\nets_waal.py\", line 56, in train\n",
      "    for index, label_x, label_y, unlabel_x, _ in loader_tr:\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\torch\\utils\\data\\dataloader.py\", line 439, in __iter__\n",
      "    return self._get_iterator()\n",
      "           ^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\torch\\utils\\data\\dataloader.py\", line 387, in _get_iterator\n",
      "    return _MultiProcessingDataLoaderIter(self)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\torch\\utils\\data\\dataloader.py\", line 1040, in __init__\n",
      "    w.start()\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\process.py\", line 121, in start\n",
      "    self._popen = self._Popen(self)\n",
      "                  ^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\context.py\", line 224, in _Popen\n",
      "    return _default_context.get_context().Process._Popen(process_obj)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\context.py\", line 336, in _Popen\n",
      "    return Popen(process_obj)\n",
      "           ^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\popen_spawn_win32.py\", line 45, in __init__\n",
      "    prep_data = spawn.get_preparation_data(process_obj._name)\n",
      "                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\spawn.py\", line 158, in get_preparation_data\n",
      "    _check_not_importing_main()\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\spawn.py\", line 138, in _check_not_importing_main\n",
      "    raise RuntimeError('''\n",
      "RuntimeError: \n",
      "        An attempt has been made to start a new process before the\n",
      "        current process has finished its bootstrapping phase.\n",
      "\n",
      "        This probably means that you are not using fork to start your\n",
      "        child processes and you have forgotten to use the proper idiom\n",
      "        in the main module:\n",
      "\n",
      "            if __name__ == '__main__':\n",
      "                freeze_support()\n",
      "                ...\n",
      "\n",
      "        The \"freeze_support()\" line can be omitted if the program\n",
      "        is not going to be frozen to produce an executable.\n",
      "\n",
      "  0%|                                                                        | 0/20 [00:00<?, ?it/s]\n",
      "  0%|                                                                        | 0/20 [00:00<?, ?it/s]\n",
      "Traceback (most recent call last):\n",
      "  File \"<string>\", line 1, in <module>\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\spawn.py\", line 120, in spawn_main\n",
      "    exitcode = _main(fd, parent_sentinel)\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\spawn.py\", line 129, in _main\n",
      "    prepare(preparation_data)\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\spawn.py\", line 240, in prepare\n",
      "    _fixup_main_from_path(data['init_main_from_path'])\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\spawn.py\", line 291, in _fixup_main_from_path\n",
      "    main_content = runpy.run_path(main_path,\n",
      "                   ^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"<frozen runpy>\", line 291, in run_path\n",
      "  File \"<frozen runpy>\", line 98, in _run_module_code\n",
      "  File \"<frozen runpy>\", line 88, in _run_code\n",
      "  File \"C:\\Users\\antig\\Documents\\GitHub\\deepALplus\\demo.py\", line 96, in <module>\n",
      "    strategy.train(model_name = args_input.ALstrategy)\n",
      "  File \"C:\\Users\\antig\\Documents\\GitHub\\deepALplus\\query_strategies\\strategy.py\", line 41, in train\n",
      "    self.net.train(labeled_data, X_labeled, Y_labeled,X_unlabeled, Y_unlabeled)\n",
      "  File \"C:\\Users\\antig\\Documents\\GitHub\\deepALplus\\nets_waal.py\", line 56, in train\n",
      "    for index, label_x, label_y, unlabel_x, _ in loader_tr:\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\torch\\utils\\data\\dataloader.py\", line 439, in __iter__\n",
      "    return self._get_iterator()\n",
      "           ^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\torch\\utils\\data\\dataloader.py\", line 387, in _get_iterator\n",
      "    return _MultiProcessingDataLoaderIter(self)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\torch\\utils\\data\\dataloader.py\", line 1040, in __init__\n",
      "    w.start()\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\process.py\", line 121, in start\n",
      "    self._popen = self._Popen(self)\n",
      "                  ^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\context.py\", line 224, in _Popen\n",
      "    return _default_context.get_context().Process._Popen(process_obj)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\context.py\", line 336, in _Popen\n",
      "    return Popen(process_obj)\n",
      "           ^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\popen_spawn_win32.py\", line 45, in __init__\n",
      "    prep_data = spawn.get_preparation_data(process_obj._name)\n",
      "                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\spawn.py\", line 158, in get_preparation_data\n",
      "    _check_not_importing_main()\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\spawn.py\", line 138, in _check_not_importing_main\n",
      "    raise RuntimeError('''\n",
      "RuntimeError: \n",
      "        An attempt has been made to start a new process before the\n",
      "        current process has finished its bootstrapping phase.\n",
      "\n",
      "        This probably means that you are not using fork to start your\n",
      "        child processes and you have forgotten to use the proper idiom\n",
      "        in the main module:\n",
      "\n",
      "            if __name__ == '__main__':\n",
      "                freeze_support()\n",
      "                ...\n",
      "\n",
      "        The \"freeze_support()\" line can be omitted if the program\n",
      "        is not going to be frozen to produce an executable.\n",
      "\n",
      "  0%|                                                                        | 0/20 [00:00<?, ?it/s]\n",
      "  0%|                                                                        | 0/20 [00:00<?, ?it/s]\n",
      "Traceback (most recent call last):\n",
      "  File \"<string>\", line 1, in <module>\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\spawn.py\", line 120, in spawn_main\n",
      "    exitcode = _main(fd, parent_sentinel)\n",
      "               ^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\spawn.py\", line 129, in _main\n",
      "    prepare(preparation_data)\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\spawn.py\", line 240, in prepare\n",
      "    _fixup_main_from_path(data['init_main_from_path'])\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\spawn.py\", line 291, in _fixup_main_from_path\n",
      "    main_content = runpy.run_path(main_path,\n",
      "                   ^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"<frozen runpy>\", line 291, in run_path\n",
      "  File \"<frozen runpy>\", line 98, in _run_module_code\n",
      "  File \"<frozen runpy>\", line 88, in _run_code\n",
      "  File \"C:\\Users\\antig\\Documents\\GitHub\\deepALplus\\demo.py\", line 96, in <module>\n",
      "    strategy.train(model_name = args_input.ALstrategy)\n",
      "  File \"C:\\Users\\antig\\Documents\\GitHub\\deepALplus\\query_strategies\\strategy.py\", line 41, in train\n",
      "    self.net.train(labeled_data, X_labeled, Y_labeled,X_unlabeled, Y_unlabeled)\n",
      "  File \"C:\\Users\\antig\\Documents\\GitHub\\deepALplus\\nets_waal.py\", line 56, in train\n",
      "    for index, label_x, label_y, unlabel_x, _ in loader_tr:\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\torch\\utils\\data\\dataloader.py\", line 439, in __iter__\n",
      "    return self._get_iterator()\n",
      "           ^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\torch\\utils\\data\\dataloader.py\", line 387, in _get_iterator\n",
      "    return _MultiProcessingDataLoaderIter(self)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\torch\\utils\\data\\dataloader.py\", line 1040, in __init__\n",
      "    w.start()\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\process.py\", line 121, in start\n",
      "    self._popen = self._Popen(self)\n",
      "                  ^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\context.py\", line 224, in _Popen\n",
      "    return _default_context.get_context().Process._Popen(process_obj)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\context.py\", line 336, in _Popen\n",
      "    return Popen(process_obj)\n",
      "           ^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\popen_spawn_win32.py\", line 45, in __init__\n",
      "    prep_data = spawn.get_preparation_data(process_obj._name)\n",
      "                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\spawn.py\", line 158, in get_preparation_data\n",
      "    _check_not_importing_main()\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\spawn.py\", line 138, in _check_not_importing_main\n",
      "    raise RuntimeError('''\n",
      "RuntimeError: \n",
      "        An attempt has been made to start a new process before the\n",
      "        current process has finished its bootstrapping phase.\n",
      "\n",
      "        This probably means that you are not using fork to start your\n",
      "        child processes and you have forgotten to use the proper idiom\n",
      "        in the main module:\n",
      "\n",
      "            if __name__ == '__main__':\n",
      "                freeze_support()\n",
      "                ...\n",
      "\n",
      "        The \"freeze_support()\" line can be omitted if the program\n",
      "        is not going to be frozen to produce an executable.\n",
      "\n",
      "  0%|                                                                        | 0/20 [00:10<?, ?it/s]\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\torch\\utils\\data\\dataloader.py\", line 1133, in _try_get_data\n",
      "    data = self._data_queue.get(timeout=timeout)\n",
      "           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\multiprocessing\\queues.py\", line 114, in get\n",
      "    raise Empty\n",
      "_queue.Empty\n",
      "\n",
      "The above exception was the direct cause of the following exception:\n",
      "\n",
      "Traceback (most recent call last):\n",
      "  File \"C:\\Users\\antig\\Documents\\GitHub\\deepALplus\\demo.py\", line 96, in <module>\n",
      "    strategy.train(model_name = args_input.ALstrategy)\n",
      "  File \"C:\\Users\\antig\\Documents\\GitHub\\deepALplus\\query_strategies\\strategy.py\", line 41, in train\n",
      "    self.net.train(labeled_data, X_labeled, Y_labeled,X_unlabeled, Y_unlabeled)\n",
      "  File \"C:\\Users\\antig\\Documents\\GitHub\\deepALplus\\nets_waal.py\", line 56, in train\n",
      "    for index, label_x, label_y, unlabel_x, _ in loader_tr:\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\torch\\utils\\data\\dataloader.py\", line 631, in __next__\n",
      "    data = self._next_data()\n",
      "           ^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\torch\\utils\\data\\dataloader.py\", line 1329, in _next_data\n",
      "    idx, data = self._get_data()\n",
      "                ^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\torch\\utils\\data\\dataloader.py\", line 1295, in _get_data\n",
      "    success, data = self._try_get_data()\n",
      "                    ^^^^^^^^^^^^^^^^^^^^\n",
      "  File \"C:\\Users\\antig\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\torch\\utils\\data\\dataloader.py\", line 1146, in _try_get_data\n",
      "    raise RuntimeError(f'DataLoader worker (pid(s) {pids_str}) exited unexpectedly') from e\n",
      "RuntimeError: DataLoader worker (pid(s) 1172, 18168, 15764, 10824) exited unexpectedly\n"
     ]
    }
   ],
   "source": [
    "!python demo.py -a WAAL -s 100 -q 1000 -b 100 -d MNIST --seed 4666 -t 3 -g 0 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9e750416-8ab9-4016-8a4c-f2e4dda5cd7f",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
